lib/activerecord-import/adapters/mysql_adapter.rb

```diff
@@ -77,7 +77,7 @@ def add_column_for_on_duplicate_key_update( column, options = {} ) # :nodoc:
       when Array then columns << column.to_sym unless columns.include?(column.to_sym)
       when Hash then columns[column.to_sym] = column.to_sym
       end
-    elsif !options[:ignore] && !options[:on_duplicate_key_ignore]
+    elsif !options[:ignore] && !options[:on_duplicate_key_ignore] && !options[:on_duplicate_raise]
```

Comment:
I just tried and it does not raise an ActiveRecord NotUnique exception on my side.. (cf my comment below about tests)